File: ros/ws02/first_ws/src/hello_pkg/scripts/hello.py
#! /usr/bin/env python
# coding=utf8

import rospy

if __name__ == '__main__':
    nodeName = "hello_node"
    rospy.init_node(nodeName)

    rate = rospy.Rate(2)
    while not rospy.is_shutdown():
        print "hello ros"
        rate.sleep()

    rospy.spin()

File: 06ros/ws02/first_ws/src/hello_pkg/src/hello.cpp
//
// Created by zcb on 5/3/21.
//
#include <iostream>
#include <ros/ros.h>


int main(int argc, char **argv) {

    std::string nodeName = "hello_node";
    ros::init(argc, argv, nodeName);
    ros::NodeHandle node;


    ros::Rate rate(2);

    while (ros::ok()) {
        std::cout << "hello ros" << std::endl;


        rate.sleep();
    }
    ros::spin();


    return 0;
}